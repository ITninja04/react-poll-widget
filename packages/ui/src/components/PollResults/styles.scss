@import '../../colors';

.poll-results {
  display: flex;
  flex-flow: column;
  font-size: 16px;
  font-weight: bold;

  .poll-result {
    position: relative;

    .ui.progress {
      margin-bottom: 0.5em;
      background: $white;

      .bar {
        height: 2.25em;
        background: rgba($darkwhite, 0.8);
      }
    }

    label {
      position: absolute;
      display: flex;
      align-items: center;

      top: 0;
      height: calc(100% - 0.5em);

      .score {
        padding: 0 0.5em;
      }
    }
  }
}