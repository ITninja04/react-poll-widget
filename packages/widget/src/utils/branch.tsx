export type BranchFunction = (condition: boolean, ComponentIfTrue: React.ReactNode, ComponentIfFalse: React.ReactNode) => React.ReactNode;

export const branch: BranchFunction = (condition, ComponentIfTrue, ComponentIfFalse) => condition
  ? ComponentIfTrue
  : ComponentIfFalse;