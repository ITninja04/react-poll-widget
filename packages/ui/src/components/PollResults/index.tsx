import React from 'react';
import { Progress } from 'semantic-ui-react';

import './styles.scss';

type PollResult = {
  label: string;
  score: number;
};

const PollResults: React.FC<{
  results: PollResult[]
}> = ({
  results
}) => (
      <div className='poll-results'>
        {
          results.map(result => (
            <div className='poll-result'>
              <Progress
                percent={result.score}
              />
              <label>
                <span className='score'>
                  {result.score}%
                </span>
                <span className="label">
                  {result.label}
                </span>
              </label>
            </div>
          ))
        }
      </div>
    );

export default PollResults;