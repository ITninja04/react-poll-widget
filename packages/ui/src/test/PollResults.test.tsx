import PollResults from "../components/PollResults";
import { createSnapshotTest } from "./utils";

createSnapshotTest(PollResults, {
  results: [
    { option: { id: '1', value: '1', label: 'Test 1' }, score: 15 },
    { option: { id: '2', value: '2', label: 'Test 2' }, score: 65 },
    { option: { id: '3', value: '3', label: 'Test 3' }, score: 25 },
    { option: { id: '4', value: '4', label: 'Test 4' }, score: 75 }
  ]
}, 'Poll results')