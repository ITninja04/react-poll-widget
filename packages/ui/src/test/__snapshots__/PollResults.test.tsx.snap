// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Snapshot) Poll results should render 1`] = `
<DocumentFragment>
  <div
    class="poll-results"
  >
    <div
      class="poll-result"
    >
      <div
        class="ui progress"
        data-percent="15"
      >
        <div
          class="bar"
          style="width: 15%;"
        />
      </div>
      <label>
        <span
          class="score"
        >
          15%
        </span>
        <span
          class="label"
        >
          Test 1
        </span>
      </label>
    </div>
    <div
      class="poll-result"
    >
      <div
        class="ui progress"
        data-percent="65"
      >
        <div
          class="bar"
          style="width: 65%;"
        />
      </div>
      <label>
        <span
          class="score"
        >
          65%
        </span>
        <span
          class="label"
        >
          Test 2
        </span>
      </label>
    </div>
    <div
      class="poll-result"
    >
      <div
        class="ui progress"
        data-percent="25"
      >
        <div
          class="bar"
          style="width: 25%;"
        />
      </div>
      <label>
        <span
          class="score"
        >
          25%
        </span>
        <span
          class="label"
        >
          Test 3
        </span>
      </label>
    </div>
    <div
      class="poll-result"
    >
      <div
        class="ui progress"
        data-percent="75"
      >
        <div
          class="bar"
          style="width: 75%;"
        />
      </div>
      <label>
        <span
          class="score"
        >
          75%
        </span>
        <span
          class="label"
        >
          Test 4
        </span>
      </label>
    </div>
  </div>
</DocumentFragment>
`;
